### PR TITLE
feat: entity select - interface and aws

### LINKED ISSUE #403: Ability to pick attributes to return from an entity query
```ts
MyEntity.query({
  pk
}, {
  pick: ["field1", "field2"]
});
```

### PR DESCRIPTION
**Pull request by thantos**
Closes #403 

* [x] interface
* [x] AWS
* [x] local
* [x] tests
* [x] docs

### CODE REVIEW

**Review by sam-goodwin — COMMENTED** [review]
Add some tests and use // @ts-expect-error to ensure the types work?

**Review by sam-goodwin — APPROVED** [review]

**Comment by sam-goodwin** [comment]
How do we make sure it major version bumps?

**Comment by thantos** [comment]
> How do we make sure it major version bumps?

I actually don't know how or why it major version bumps...

**Comment by sam-goodwin** [comment]
Is it:

> BREAKING CHANGE: (description)

**Comment by thantos** [comment]
> Is it:
> 
> > BREAKING CHANGE: (description)

Its based on conventional commits, so its mostly been working off of feat/fix tags, but BREAKING CHANGE will also force it.

https://github.com/lerna/lerna/tree/main/libs/commands/version#--conventional-commits

**Comment by thantos** [comment]
Though I am unsure what it will do when the version is pre-1

**Comment by sam-goodwin** [comment]
> Though I am unsure what it will do when the version is pre-1

No biggie. We can just move forward.

File: apps/tests/aws-runtime/test/tester.test.ts [hunk 1]
```diff
@@ -226,14 +226,14 @@ eventualRuntimeTestHarness(
...
+        { counterNumber: 2, n: 2 },
+        { counterNumber: 3, n: 1 },
       ],
       expect.arrayContaining([
-        [3, 1],
-        [1, 1],
-        [2, 2],
+        { counterNumber: 1, n: 1 },
+        { counterNumber: 2, n: 2 },
+        { counterNumber: 3, n: 1 },
```
Comment (sam-goodwin) [inline apps/tests/aws-runtime/test/tester.test.ts:236]: What changed?

Comment (thantos) [reply apps/tests/aws-runtime/test/tester.test.ts:236]: I changed this to use `select` instead of mapping to an array.

**Pull request merged**